feat(bulk-import): allow adding repositories

debsmita1 opened the pull request:
**Resolves:**
https://issues.redhat.com/browse/RHIDP-1664

**GIF:**

https://github.com/janus-idp/backstage-plugins/assets/22490998/e9f8e67d-63e3-426b-88ff-c6708c0917a7

ShiranHi commented:
@debsmita1 it looks good! I am happy to see that! 🙌

I have few questions: 
1. Can we increase the width of the segmented button "Repositories" and "Organization"?
2.  If the status is "Failed generating file" we want to add a "Try again" button, and if users uncheck and checked again we'll try to generate it again ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=72%3A1254&mode=design&t=stWEmLJBJ9WuLCFs-1))
3. If ServiceNow selected the button label will be “Create ServiceNow tickets” instead of "Create pull requests"
4. btw, cam we change the label to be based on the number of repositores selected? if it is just one then the button label will be "Create pull request" (and not requests)? same question about "Create ServiceNow ticket"
5. I saw in the video that the title says "Selected repositories (7)" but there is one organization with 3 selected repositories and one with 2 selected repositories, should it be "Selected repositories (5)"?
6. When on the "Organization" view, I see one of them with the "Repository already added", I think we should not show that on the Repositories view unless all repositories from this organization added
7. If the repository already added the checkbox should be checked and disabled ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=11%3A4603&mode=design&t=gpY5PwhTjZYCJDVN-1))
8. Can we make the bottom bar with the buttons sticky to the bottom so users will always see it?

debsmita1 commented:
> 6. I saw in the video that the title says "Selected repositories (7)" but there is one organization with 3 selected repositories and one with 2 selected repositories, should it be "Selected repositories (5)"?

The tables have mock data as of now, As part of this PR I am only focusing on creating the tables and the components. But we should be able to handle it in this ticket https://issues.redhat.com/browse/RHIDP-1665

> 7. When on the "Organization" view, I see one of them with the "Repository already added", I think we should not show that on the Repositories view unless all repositories from this organization added

Should I leave it blank for cases where all the repositories are not added ? or Should I show "Not generated ?

> 10. Below the "Bulk import" side navigation item there is a border line, I think we don't need it

The divider was already present below  the Tech Radar. I simply grouped the Tech Radar and the Bulk Import nav items together

ShiranHi commented:
> > 13. When on the "Organization" view, I see one of them with the "Repository already added", I think we should not show that on the Repositories view unless all repositories from this organization added
> 
> Should I leave it blank for cases where all the repositories are not added ? or Should I show "Not generated 
> 
If no respiratory from this org added then we will write "Not generated". If some added, let's write "x ready", if repositories from this org added, then we will write "Ready". What do you think?

> > 10. Below the "Bulk import" side navigation item there is a border line, I think we don't need it
> 
> The divider was already present below the Tech Radar. I simply grouped the Tech Radar and the Bulk Import nav items together
Got it, I think we don't need it

debsmita1 commented:
> > > 13. When on the "Organization" view, I see one of them with the "Repository already added", I think we should not show that on the Repositories view unless all repositories from this organization added
> > 
> > 
> > Should I leave it blank for cases where all the repositories are not added ? or Should I show "Not generated
> 
> If no respiratory from this org added then we will write "Not generated". If some added, let's write "x ready", if repositories from this org added, then we will write "Ready". What do you think?
> 

This makes sense

ciiay reviewed (commented):
Can you confirm if `yarn start` works in the `plugins/bulk-import` directory? The previous version of this PR worked for me but this one doesn't.  After pulling upstream/main branch code and run `yarn install` `yarn start` it worked. The feature itself works good.

ciiay commented on plugins/bulk-import/src/components/AddRepositories/RepositoriesTable.tsx:
```diff
@@ -0,0 +1,258 @@
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelected = data
```
Just a question, if user filtered data then click select all checkbox, will it select all data or only filtered data?

debsmita1 replied:
Fixed it


https://github.com/janus-idp/backstage-plugins/assets/22490998/321443e5-960b-4839-bf0b-e8217cee63b4

invincibleJai commented:
@debsmita1 e2e tests are failing

debsmita1 commented:
> Can you confirm if `yarn start` works in the `plugins/bulk-import` directory? The previous version of this PR worked for me but this one doesn't. After pulling upstream/main branch code and run `yarn install` `yarn start` it worked. The feature itself works good.

Hey @ciiay ! It is working for me. Could you please try again ?

ciiay commented:
@debsmita1 `yarn start` in `plugins/bulk-import/` now works for me too.

ciiay reviewed (approved):
LGTM
Thank you 👍

invincibleJai commented on plugins/bulk-import/package.json:
```diff
@@ -5,7 +5,7 @@
   "license": "Apache-2.0",
   "publishConfig": {
-    "access": "public",
+    "access": "restricted",
```
i don't think this is needed, you might want to add https://github.com/janus-idp/backstage-plugins/blob/main/plugins/argocd/package.json#L7

invincibleJai commented on plugins/bulk-import/src/components/AddRepositories/AddRepositoriesPage.tsx:
```diff
@@ -0,0 +1,162 @@
+            </div>
+            <div
+              className="bs-bi-illustrations"
+              style={{
+                display: 'inline-block',
+                verticalAlign: 'top',
+                textAlign: 'center',
+                fontSize: '16px',
+                marginLeft: '50px',
+              }}
```
nit: can we avoid inline styles?

invincibleJai commented on plugins/bulk-import/src/components/AddRepositories/AddRepositoriesTable.tsx:
```diff
@@ -0,0 +1,58 @@
+    shouldValidate?: boolean,
+  ) => Promise<FormikErrors<AddRepositoriesFormValues>> | Promise<void>;
+}) => {
+  const [searchString, setSearchString] = React.useState<string>('');
+  const [page, setPage] = React.useState(0);
```
nit: can we add types for this as well?

invincibleJai commented on plugins/bulk-import/src/components/Repositories/RepositoriesList.tsx:
```diff
@@ -20,17 +21,20 @@ export const RepositoriesList = () => {
+    <>
+      <RepositoriesListToolbar />
+      <Table
+        title="Added repositories (0)"
```
should this `0` be always 0?

invincibleJai reviewed (commented):
Thanks @debsmita1 , can we add unit tests as well as part of this PR?

invincibleJai commented:
Here are some observations

- does CTA has to be sticky?  in dev mode it shows like below (and there are lot of empty space on right of first collapsible card)

<img width="1722" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/cbf7f2cb-960e-4125-a63f-6a4ef660e1b5">

- on clicking cancel on above CTA i see blank screen on dev mode

<img width="1726" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/e060a0f7-40d4-4edd-9abe-b7ad5e4805c5">

- Getting below warning

<img width="1725" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/945cc29a-f1bb-4e3b-b82e-32127fe8a400">

ShiranHi commented:
@debsmita1 it looks good, that was fast 🙂

Few questions:
1. I got feedback on the top collapsable banner and we want to change the illustrations there ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=11%3A4603&mode=design&t=xXG6x9THlQgmk9Rm-1))
2. Let's try to horizontally align the icons there so they will fit the card width (as in the design)
3. We had a micro-copy review and the title of the card has changed to be "Add repositories to Red Hat Developer Hub in 5 steps" and the description text slightly changed too 
4. The bottom bar with the sticky buttons - can we make it higher? So we'll have 24px above and below it?
5. Tooltip appears on hover - can we prevent having a second line just for one word?
6. I saw on the first row that we write "Generating..." and there is "Try again" button next to it. I think we should not have scenario like this
7. If the status is "Failed generating file" and users check it again, we should re-try to create it, so after checking this repo, we should say "Generating..."

debsmita1 commented:
> Here are some observations
> 
> * does CTA has to be sticky?  in dev mode it shows like below (and there are lot of empty space on right of first collapsible card)
> 

Yes, the CTA has to be sticky. Fixed the empty space on the right

> <img alt="image" width="1722" src="https://private-user-images.githubusercontent.com/5129024/319448747-cbf7f2cb-960e-4125-a63f-6a4ef660e1b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTUwOTUsIm5iZiI6MTcxMjI1NDc5NSwicGF0aCI6Ii81MTI5MDI0LzMxOTQ0ODc0Ny1jYmY3ZjJjYi05NjBlLTQxMjUtYTYzZi02YTRlZjY2MGUxYjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDRUMTgxOTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmU0OTZkN2MxMTc3MjA4OGY0ZmRiYjhlNDVkNDAzODhhMmVmNDI0YWViMDljN2Q3MmQ2ODJmNzA2YjIyZWRlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.puTunYyd0TkzuZxY2D-KLEYVv0tXyvuUCrM1kOfXZY4">
> * on clicking cancel on above CTA i see blank screen on dev mode
> 

> <img alt="image" width="1726" src="https://private-user-images.githubusercontent.com/5129024/319449479-e060a0f7-40d4-4edd-9abe-b7ad5e4805c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTUwOTUsIm5iZiI6MTcxMjI1NDc5NSwicGF0aCI6Ii81MTI5MDI0LzMxOTQ0OTQ3OS1lMDYwYTBmNy00MGQ0LTRlZGQtOWFiZS1iN2FkNWU0ODA1YzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDRUMTgxOTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDMzNWE3MDlhMDg2NzRjZTUzNGI0ODdhNDcxZWE0ZWVmNjgxMTYyZDQyMTg0MzU2NmM3ZDkyY2Y2MTI4OWZhOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D5_tiOEZPtyLkX_SWaNvYEuXHIc1EM_1FbyUDD038kc">
> * Getting below warning
> 

Fixed it



> <img alt="image" width="1725" src="https://private-user-images.githubusercontent.com/5129024/319449015-945cc29a-f1bb-4e3b-b82e-32127fe8a400.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTUwOTUsIm5iZiI6MTcxMjI1NDc5NSwicGF0aCI6Ii81MTI5MDI0LzMxOTQ0OTAxNS05NDVjYzI5YS1mMWJiLTRlM2ItYjgyZS0zMjEyN2ZlOGE0MDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDRUMTgxOTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmI1YmNkMWRhMGQwOGQyZjA4YmMwOGNhYjRiMDlhMDY3ZDNjNzhlZmQxNDc1OWQ2NTAzZjNiNDliZmI5YjE0MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Wy7z8Ma_MvaP3Cp843JVPxMFbyNOMPMJ6ynrQfwUZiA">

on it

debsmita1 commented:
> @debsmita1 it looks good, that was fast 🙂
> 
> Few questions:
> 
> 1. I got feedback on the top collapsable banner and we want to change the illustrations there ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=11%3A4603&mode=design&t=xXG6x9THlQgmk9Rm-1))
> 2. We had a micro-copy review and the title of the card has changed to be "Add repositories to Red Hat Developer Hub in 5 steps" and the description text slightly changed too
> 3. The bottom bar with the sticky buttons - can we make it higher? So we'll have 24px above and below it?
> 4. Tooltip appears on hover - can we prevent having a second line just for one word?
> 5. I saw on the first row that we write "Generating..." and there is "Try again" button next to it. I think we should not have scenario like this
> 6. If the status is "Failed generating file" and users check it again, we should re-try to create it, so after checking this repo, we should say "Generating..."
> 7. Now that I see it, the "Approval tool" is quite disappearing, do you mind moving them to the top of the page? I hope we have time for this improvement ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=392%3A599&mode=design&t=xXG6x9THlQgmk9Rm-1))

Updated the GIF. PTAL

ciiay commented on plugins/bulk-import/src/components/AddRepositories/AddRepositoriesPage.tsx:
```diff
@@ -0,0 +1,64 @@
+              Add repositories to Red Hat Developer Hub in 5 steps
+            </Typography>
+          </AccordionSummary>
+          <AccordionDetails id="add-repository-illustrations">
```
We can use `display: flex` in the Illustrations container and use `justify-content: space-around` instead of hard coded right padding of illustrations.
<img width="1285" alt="rhidp_1664_pr_review" src="https://github.com/janus-idp/backstage-plugins/assets/26255262/a0d9da85-5932-4962-a774-e9cc83db2a6e">

debsmita1 replied:
Adding `display: flex` to the container wrapping the illustrations, the mobile view looks like this
<img width="1069" alt="Screenshot 2024-04-05 at 10 41 06 AM" src="https://github.com/janus-idp/backstage-plugins/assets/22490998/0ce60eb4-3b84-425c-ad7c-5bed551ec757">

ciiay replied:
Oh I didn't know we need to deal with mobile view as well. In that case is adding a `overflow: scroll` acceptable? The current padding doesn't work on my laptop default screen. There might be other displays we need to review and get some layout design from UX team for better mobile screen display.

https://github.com/janus-idp/backstage-plugins/assets/26255262/0ef6706e-a585-4321-b51a-7a17b8d689da

ShiranHi commented:
@debsmita1 thank you for the last minute changes, it looks much better now 🙌

Two questions:
1. The line below the "Approval tool" seems too bright, can we make sure it is the same as in the design ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=392%3A599&mode=design&t=GFVjQ7qpvPeDYLAz-1))?
2. Sticky buttons - do they have a transparent background? Cab we make sure it is white background and has a line above it?

invincibleJai commented:
- In dev mode i see below warnings, check if this is relevant

<img width="1728" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/4e95caa5-ebc0-4a01-9aa5-9657c927e8c2">

<img width="1724" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/de2f39cc-5774-456a-9caa-93fa509bffc1">

- on the right in the collapsible panel I see it has more empty space and even figma has some @ShiranHi is this expected?

<img width="1652" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/1672948f-8a26-4679-a861-736e4486a46d">

Figma:

<img width="1017" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/6413c228-25b2-48a9-b443-12358b2aa821">

debsmita1 commented:
> @debsmita1 thank you for the last minute changes, it looks much better now 🙌
> 
> Two questions:
> 
> 1. The line below the "Approval tool" seems too bright, can we make sure it is the same as in the design ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=392%3A599&mode=design&t=GFVjQ7qpvPeDYLAz-1))?
> 2. Sticky buttons - do they have a transparent background? Cab we make sure it is white background and has a line above it?

@ShiranHi  the background is not transparent

<img width="1561" alt="Screenshot 2024-04-08 at 8 10 09 PM" src="https://github.com/janus-idp/backstage-plugins/assets/22490998/fcc0b679-5b14-4f1b-9fd9-ae10e954789d">

ciiay reviewed (commented):
The According works great on my end now. Thanks for the update.
There appears a line above bottom buttons which wasn't there before. Is this added on purpose?

ShiranHi commented:
> * on the right in the collapsible panel I see it has more empty space and even figma has some @ShiranHi is this expected?
Yes, that is ok. It will be great if we can have some padding on the left side, just before the first icon. 

>The line below the "Approval tool" seems too bright, can we make sure it is the same as in the design ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=392%3A599&mode=design&t=GFVjQ7qpvPeDYLAz-1))?
Thank you for fixing this, now it seems too dark, can we make sure it is has the same gray color as the segmented button border (with the Repository and the Organization buttons)?

>@ShiranHi the background is not transparent
Can we have a White background there and that the line will not have padding on the right side?
<img width="400" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/77277517/d9e9c2d8-ada2-40ca-8220-ce80d97b785f">

ciiay commented:
>Can we have a White background there and that the line will not have padding on the right side?

Hi @ShiranHi @debsmita1 The white background setting will be addressed as default main container background color in [this PR](https://github.com/janus-idp/backstage-showcase/pull/1171)  on backstage-showcase.

debsmita1 commented:
> > * on the right in the collapsible panel I see it has more empty space and even figma has some @ShiranHi is this expected?
> 
> Yes, that is ok. It will be great if we can have some padding on the left side, just before the first icon.
> 
> > The line below the "Approval tool" seems too bright, can we make sure it is the same as in the design ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=392%3A599&mode=design&t=GFVjQ7qpvPeDYLAz-1))?
> 
> Thank you for fixing this, now it seems too dark, can we make sure it is has the same gray color as the segmented button border (with the Repository and the Organization buttons)?
> 
> > @ShiranHi the background is not transparent
> 
> Can we have a White background there and that the line will not have padding on the right side? <img alt="image" width="400" src="https://private-user-images.githubusercontent.com/77277517/320716645-d9e9c2d8-ada2-40ca-8220-ce80d97b785f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MDc4NTEsIm5iZiI6MTcxMjgwNzU1MSwicGF0aCI6Ii83NzI3NzUxNy8zMjA3MTY2NDUtZDllOWMyZDgtYWRhMi00MGNhLTgyMjAtY2U4MGQ5N2I3ODVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDExVDAzNTIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwYzA2NDBhNDUwN2I1MjIwNTY2MzQ5OTA2NzkwZTc1NWU2ZjRiMmE0Y2U4Y2Y5YzQwZGJkZGE4YmY0MWI4MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mFKRN0ndWgodrYAiPOA1duP4i7HW_kYIh_FCn5N1I5g">

@ShiranHi Updated , PTAL
<img width="1561" alt="Screenshot 2024-04-11 at 9 23 51 AM" src="https://github.com/janus-idp/backstage-plugins/assets/22490998/2df94c78-f23d-4caf-b982-9fd01df45e9c">
 
 @ciiay ^

ShiranHi commented:
@debsmita1 looks good to me, thank you!

invincibleJai commented:
in dark theme this looks like this

<img width="1655" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/1f381244-9f47-4544-a553-1b208d7d6f91">

@debsmita1 is this expected?

cc @ShiranHi

ShiranHi commented:
> in dark theme this looks like this
> 
> <img alt="image" width="1655" src="https://private-user-images.githubusercontent.com/5129024/321563492-1f381244-9f47-4544-a553-1b208d7d6f91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MjgxMTcsIm5iZiI6MTcxMjgyNzgxNywicGF0aCI6Ii81MTI5MDI0LzMyMTU2MzQ5Mi0xZjM4MTI0NC05ZjQ3LTQ1NDQtYTU1My0xYjIwOGQ3ZDZmOTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMDkzMDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzhlMjQxOWQyZWY5OTNkZmNlZjUzOWE3ZTkyYmJlMzZhNWFlMDllZWRlNzMyMTY5NzkwMTYxZDU0ZWY1MWZiZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IFtQICbmdewQHnhoXL1l3YFTfc4A_IWjRcHtaGnWkE4">

Right, good catch! We need to use the white version for dark theme ([icons are here](https://redhat.brand-portal.adobe.com/mediaportal.html/content/dam/mac/redhat/brand/illustrations/mini-illustrations))

debsmita1 commented on plugins/bulk-import/src/components/AddRepositories/AddRepositoriesForm.tsx:
```diff
@@ -0,0 +1,99 @@
+          </Typography>
+          <Tooltip
+            placement="top"
+            title="When adding a new repository, it requires approval. Once it's approved and the PR is approved or the ServiceNow tickets closed, the repositories will be added to the Catalog page."
+          >
```
yes @invincibleJai 
<img width="746" alt="Screenshot 2024-04-11 at 3 06 20 PM" src="https://github.com/janus-idp/backstage-plugins/assets/22490998/38d5f974-abe8-49e3-88ec-e98bc57037cf">

@ShiranHi  ^

ShiranHi replied:
> this text is as per design? got confused with
> 
> `Once it's approved and the PR is approved`

oops, that is a mistake, this is the text:
When adding a new repository, it requires approval. Once the PR is approved or the ServiceNow ticket is closed, the repositories will be added to the Catalog page.

debsmita1 replied:
Done

![Screenshot 2024-04-11 at 3 42 59 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/78c1554b-0303-4246-ae14-f4310c75ef3e)

ShiranHi commented:
> <img alt="image" width="870" src="https://private-user-images.githubusercontent.com/5129024/321566016-9eea2919-9c79-4e3b-9aae-bfbb4f133316.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4Mjk1NDMsIm5iZiI6MTcxMjgyOTI0MywicGF0aCI6Ii81MTI5MDI0LzMyMTU2NjAxNi05ZWVhMjkxOS05Yzc5LTRlM2ItOWFhZS1iZmJiNGYxMzMzMTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMDk1NDAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yjc5YzFjYzYwMzMxNzhkMWU3MWM3NWEwYmE1YWM0MmNmMDI1YmE0MDQyYmU5MWEyNDE1OWRmN2EyYmJjZmU4NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mtnfoSYTwxvTqmolWw2ypWa4bDR757YrJwuHu5DtFYA">

Remove the "Please" word at the beginning, thanks!

debsmita1 commented:
> in dark theme this looks like this
> 
> <img alt="image" width="1655" src="https://private-user-images.githubusercontent.com/5129024/321563492-1f381244-9f47-4544-a553-1b208d7d6f91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MzA3NDYsIm5iZiI6MTcxMjgzMDQ0NiwicGF0aCI6Ii81MTI5MDI0LzMyMTU2MzQ5Mi0xZjM4MTI0NC05ZjQ3LTQ1NDQtYTU1My0xYjIwOGQ3ZDZmOTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTAxNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM2OTAzOWZjNzE2OWMyZGY1YTM1YjRlMDZjMjFkOTM3NzFiZTk3MGY3NmFjYTc3MDEyYzI3MDQxMTU3MmVmZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.5VYeAjJ0dt0gtmrRiOcl67PbVG13FtDze72agUHeXdU">
> and click on repo / org open's in new window, which is expected should we have `launch` icon as we have designs <img alt="image" width="1610" src="https://private-user-images.githubusercontent.com/5129024/321564980-5c2f67c2-b41c-410b-b5ef-5115fcf0e1e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MzA3NDYsIm5iZiI6MTcxMjgzMDQ0NiwicGF0aCI6Ii81MTI5MDI0LzMyMTU2NDk4MC01YzJmNjdjMi1iNDFjLTQxMGItYjVlZi01MTE1ZmNmMGUxZTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTAxNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODg1ZGRkZWY3ZGJhYTQ0YzA2MTk5N2E1Mjc1MGYzNzE4NzkwYmViOWQyOGQ4YWM5OTE1OGU1ZWY0NDJkMDU0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qJXTjUONt3RvZ3Lf37yPxuNsSxhV_u2z79pkHZetZ6U">
> 
> <img alt="image" width="870" src="https://private-user-images.githubusercontent.com/5129024/321566016-9eea2919-9c79-4e3b-9aae-bfbb4f133316.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MzA3NDYsIm5iZiI6MTcxMjgzMDQ0NiwicGF0aCI6Ii81MTI5MDI0LzMyMTU2NjAxNi05ZWVhMjkxOS05Yzc5LTRlM2ItOWFhZS1iZmJiNGYxMzMzMTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTAxNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTZhOGE3ZjQ5MGM2OTBlMjE1OTZlYjZhNzBlMzZmMDAyMTBlZTViMDQ0NDMyNzZhZTJjMTIxNzU3NGIxZGIxNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Dmcaanpu778wiiR0r2K5sb6k_9eYPuATbXwa3Y49PFU">
> @debsmita1 ^^
> 
> cc @ShiranHi



Done

![Screenshot 2024-04-11 at 3 42 59 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/235451a8-3a7b-4aba-8c58-98e974f27f8a)

@invincibleJai @ShiranHi

invincibleJai commented:
/lgtm
/approve
